### DISCUSSION TITLE
How to make user-specific use cases in map more accessible for the user?

### DISCUSSION THREAD

**Discussion by moving-bits** [cgeo/cgeo#16185] [General]
In #15847 and the related PR #16127 we started a discussion on how to better organize map menus, to make it easier for users to find what they are looking for in that wealth of settings. We quickly discovered that the use cases vary widely, leading to different needs of which settings needs to be quickly accessible. What one user may regard as "not important for me" may be quite regularly used by some other user and vice versa.

In the past we had discussed some mechanism of full customization of menus, to be able to hide settings a user does not use. Problem with that approach was that it was regarded as too technical. More technical affine users may get the best out of this, but perhaps 99% of the users would be overwhelmed by such a feature.

Therefore I would like to propose a different mechanism, specifically for the maps context: Favorites

We already have the map quick settings, which try to offer some most-used functions at the tip of a finger. Though we may discuss if everything that's currently in there really has the importance we have thought back then that it would (eg.: compact icon mode), this approach generally has been working well, so let's expand on that.

Proposal:
- Add a "favorite" function to map quick settings, positioned at the end of current dialog.
- In factory state, this would just be a "plus" button, to add some favorite.
- For every favorite the user adds, an additional icon would appear in front of the "plus" symbol. All icons side by side in a single line.
- Tapping the "plus" button would open a list of all menu entries the map menu has, including all menu levels. Tapping on an item would add this item to the "favorite bar" (= icon row in map quick settings bottom sheet)
- To remove a favorite, long-tapping on that item could either remove it directly or open a confirmation dialog for removal.
- For selecting an item we could reuse the dialog type we have for selecting quick launch items on home screen or selecting info items for cache lists.

Possible extension:
- Not only list the menu items, but also offer items from theme options dialog. Challenge would be that those are theme-specific.

(I can create a visual mockup later this day.)

**Comment by ztNFny** [comment]
I'd rather have well organized menus than put even more "random" options in the already chaotic quick settings dialog. That one already contains so many thrown together things today.
C:geo should be optimized for the "average" user, it's simply impossible to take every single special usecase of specific persons into account in the same way.

(If 1 dev wants to have their favourite thing in a special place it's easy for you to do that, just run your custom builds).
